src/engine-actions/get-chain-transactions.js

```diff
@@ -0,0 +1,190 @@
+
+  if (!Array.isArray(transactions) || !transactions.length) {
+    this.logger.debug('No transactions exist on the engine')
+    return []
+  }
+
+  // Get all channels from the engine (rough)
+  const [
+    { channels: rawChannels },
+    { channels: rawClosedChannels },
+    {
+      pendingOpenChannels: rawPendingOpenChannels,
+      pendingClosingChannels: rawPendingClosingChannels,
+      pendingForceClosingChannels: rawPendingForceClosingChannels,
+      waitingCloseChannels: rawWaitingCloseChannels
+    }
+  ] = await Promise.all([
+    listChannels({ client: this.client }),
+    listClosedChannels({ client: this.client }),
+    listPendingChannels({ client: this.client })
+  ])
+
+  const openChannels = [
+    ...rawChannels,
+    // Pending channels have a different structure than our typical Channel object
+    // so we need to grab the channel off of the raw payload
+    ...rawPendingOpenChannels.map(chan => chan.channel)
+  ]
+
+  const closedChannels = [
+    ...rawClosedChannels,
+    // Pending channels have a different structure than our typical Channel object
+    // so we need to grab the channel off of the raw payload
+    ...rawPendingClosingChannels.map(chan => chan.channel),
+    ...rawPendingForceClosingChannels.map(chan => chan.channel),
+    ...rawWaitingCloseChannels.map(chan => chan.channel)
+  ]
+
+  const pendingChannels = [
+    ...rawPendingOpenChannels.map(chan => chan.channel),
+    ...rawPendingClosingChannels.map(chan => chan.channel),
+    ...rawPendingForceClosingChannels.map(chan => chan.channel),
+    ...rawWaitingCloseChannels.map(chan => chan.channel)
+  ]
+
+  const [
+    closedChannelTxIds,
+    closedChannelFees,
+    openChannelTxIds,
+    pendingChannelTxIds
+  ] = await Promise.all([
+    getTxIdsForChannels(closedChannels),
+    getFeesForClosedChannels(rawClosedChannels),
+    getTxIdsForChannels(openChannels),
+    getTxIdsForChannels(pendingChannels)
+  ])
+
+  return transactions.map(({ txHash, amount, blockHeight, timeStamp, totalFees }) => {
+    const isClosedChannel = closedChannelTxIds.has(txHash)
+    const isChannel = (openChannelTxIds.has(txHash) || closedChannelTxIds.has(txHash))
+    const type = determineTransactionType(amount, isChannel, isClosedChannel)
+
+    // Determine is the transaction is pending. Having a blockHeight of 0 means
+    // that the transaction has not yet been placed into a block (no confirmations)
+    const isPending = (Big(blockHeight).eq(0) || pendingChannelTxIds.has(txHash))
+
+    // `totalFees` will be zero if this is a CHANNEL_CLOSE transaction type
+    let fees = Big(totalFees)
+
+    // If we've identified the type of transaction as channel close then `totalFees`
+    // will be 0. We calculate the fee based on information from the closing transaction
+    // but this is only available if the transaction has hit the blockchain
+    if (type === TRANSACTION_TYPES.CHANNEL_CLOSE && !isPending) {
```

Comment:
don't you need to check if we were the initiator?

Reply:
@treygriffith We do need to check for the initiator, however we don't currently have the information available from LND. Ive opened a PR to add initiator to the closed channel summary, however for now this should be OK

https://github.com/sparkswap/lnd/pull/34

Reply:
Going to add a TODO to this code